its/src/test/java/com/sonar/it/scanner/msbuild/utils/BuildCommand.java

```diff
@@ -83,9 +83,13 @@ private Command createCommand() {
         // Using the msbuild command from the dotnet CLI allows to use the same parameters as the Windows version
         // https://learn.microsoft.com/en-us/dotnet/core/tools/dotnet-build#msbuild
         : Command.create("dotnet").addArgument("msbuild");
-      command.addArgument("/t:Restore,Rebuild");
+      command
+        .addArgument("/t:Restore,Rebuild")
+        .addArgument("/warnaserror:AD0001");
     } else {
-      command = Command.create("dotnet").addArgument(dotnetCommand);
+      command = Command.create("dotnet")
+        .addArgument(dotnetCommand)
+        .addArgument("-warnaserror:AD0001");
     }
     command.addArgument("-nodeReuse:false");
```

Comment:
`nodeReuse` is ignored by `dotnet build` according to this discussion:
https://github.com/dotnet/sdk/issues/40930#issuecomment-2115643919
The argument should be set inside the ifs:
dotnet: `--disable-build-servers`
msbuild: `-nodeReuse:false`